src/mlpack/methods/ann/layer/transposed_convolution.hpp

```diff
@@ -397,8 +342,11 @@ class TransposedConvolution
   //! Locally-stored transformed gradient parameter.
   arma::cube gradientTemp;
 
-  //! Locally-stored padding layer.
-  Padding<>* padding;
+  //! Locally-stored padding layer for forward propagation.
+  Padding<>* paddingForward;
```

Comment:
We should just use `Padding<> paddingForward;` instead of a pointer here, especially if we don't have a Deconstructor that deletes the object.

Reply:
Sure, it is hard to find time now because of exams but I will make the changes as soon as I find the time. Also, The `convolution` and `atrous_convolution` layers are also using pointers, should I make a change there too?

Reply:
No worries, best of luck with your exams, would be great if you can handle the `convolution` and `atrous_convolution` part as well.

Reply:
You can refer #1985, In that the other two layers are already done. :+1: . It will be great if you make this small change as fast as possible.

Reply:
I'll work on it tonight and push new commits as soon as possible